src/php.php

```diff
@@ -0,0 +1,84 @@
+<?php
+function getConnected() : ?PDO
+{
+    try
+    {
+        $connection = new PDO(
+            'mysql:host=DB;dbname=collectorapp',
+            'root',
+            'password'
+        );
+        $connection->setAttribute(PDO:: ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+        return $connection;
+    } catch (PDOException $exception)
+    {
+        echo 'connection error: ' . $exception->getMessage();
+        return null ;
+    }
+}
+
+function fetchPigmentData() : array
+{
+    // Establish a database connection
+    $connection = getConnected();
+
+    try {
+        $query = $connection->prepare('SELECT pigment.id, pigment.name, pigment.HEX, pigment.mineral, pigment.chemical, pigment.description, pigment.image_closeup, color.name AS color_name
+            FROM pigment
+            INNER JOIN color color ON pigment.color_id = color.color_id');
+        $query->execute();
+        return $query->fetchALL();
+
+    } catch (PDOException $exception){
+        // In case of a database-related error, catch the exception and print an error message
+        echo 'fetch error:' . $exception->getMessage();
+        // Return an empty array if there's an error
+        return[];
+    }
+}
+function createpigmaDIVS(array $collection = null): string
+{
+    if (empty($collection) || !is_array($collection)) {
+
+
+        return '<div>Invalid pigment data.</div>';
+    }
+    $result = "";
+
+    foreach ($collection as $pigment) {
+        if(!isset($pigment['id']) || is_null($pigment['id'])){
+            return '';
+        }
+        if(!is_int($pigment['id'])){
+            throw new UnexpectedValueException('id was not a num');
```

Comment:
```suggestion
            throw new UnexpectedValueException('id was not a number');
```
We generally avoid shortening words because it quickly gets confusing when you have to read this error message. The infamous example is when people shorten 'category' to 'cat' - short for category? something to do with animals? an acronym? It's unnecessary to over-shorten your words.